packages/react/src/components/Storage/FileUploader/UploadTracker/UploadTracker.tsx

```diff
@@ -67,9 +67,9 @@ export function UploadTracker({
 
   useEffect(() => {
     if (shouldAutoLoad && fileState === FileState.INIT) {
-      startUpload();
+      handleUploadFile();
     }
-  });
+  }, [shouldAutoLoad, fileState, handleUploadFile]);
```

Comment:
Not concerned abt this immediately since it's likely that this effect will be refactored out, but this reads like there is potential that this could run twice for the same file

Reply:
I'll keep an eye on this during the useReducer refactor